### PR TITLE
[Button] Prefer user key events over those handled by Fluent

### PR DESCRIPTION
**Pull request by nakambo**
### Platforms Impacted
- [ ] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [ ] android

### Description of changes

Currently, on Windows, the `Button` component handles keyUp events (keyDown for macOS). What that means is someone can't have a keyUp (or keyDown) handler of their own to do "special" things on keypresses on buttons (e.g. perhaps implement a split button).

Previously (prior to PR https://github.com/microsoft/fluentui-react-native/pull/1373), `Button` always used keyUp, so button users could at least use keyDown to do things, but now depending on the target platform, a button user would have to handle the slot NOT used by Fluent.

I investigated the possibility of creating a composite handler (such that for user requested keys we always pass back to the user, and if not specified, then we pass back the handling to Fluent). It looks like it will need substantial refactoring because the event types for key handlers in useKeyProps does not match that exposed by Win32. Further, the approach to specifying valid keys (the name of the prop, as well the specifics like modifiers) is different for the Mac.

Therefore, in this change, I am instead suggesting the *other* approach I previously discussed with Saad -- if a button user tries to override a particular key handler, the user is also responsible for providing the default behaviors (like Enter\Space to invoke the button event).

For client usage correctness (and convenience), also expose a flag to indicate platform behavior with respect to keyUp or keyDown.

### Verification

Manually, using Fluent tester.

* Regular button use case (no key handler): Enter\Space works as expected
* New button added to handle key events: ONLY handled key events pass through; since Enter\Space isn't one of them, this lets us prevent default action (of course, handling those two keys manually in the custom handler would restore that behavior).

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [x] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by rurikoaraki** [comment]
For my own understanding, the aim here is to make it so that if a custom key handler is passed in, then it ignores the built in functionality completely?

**Comment by nakambo** [comment]
> For my own understanding, the aim here is to make it so that if a custom key handler is passed in, then it ignores the built in functionality completely?

Indeed, this is what Saad and I previously discussed. We liked this option because it was simpler, since it didn't require reconciling two sets of event listeners (Fluent and the actual component user\client) and possibly notifying both of them\not notifying keys to the listener that didn't request that key.

**Comment by Saadnajmi** [comment]
@Ruriko for more context, a nicer solution would be to have all RN desktop platforms respect `e.preventDefault()` so that client code could choose programmatically to suppress the default behavior. Doing so requires a bunch of native changes to React Native macOS so it's harder. Thanks for making the fix @nakambo !

File: packages/utils/interactive-hooks/src/useKeyProps.ts
Comment (Saadnajmi) [inline packages/utils/interactive-hooks/src/useKeyProps.ts:60]: Where do you actually read and use this? In client code?

Comment (nakambo) [reply packages/utils/interactive-hooks/src/useKeyProps.ts:60]: That's the idea -- it could be used in the client code. I plan to handle my key + Enter + Space in the handler indicated by this boolean, but in theory, someone could do what I was previously doing, viz. use this flag to use the OTHER slot that isn't used by Fluent.

**Comment by nakambo** [comment]
> For my own understanding, the aim here is to make it so that if a custom key handler is passed in, then it ignores the built in functionality completely?

@rurikoaraki, so this is indeed a breaking change to (already broken) clients -- because if they don't already handle Enter\Space for example, they'll need to now (previously they were broken because their custom keys wouldn't be handled, now they'd be broken for a different reason).

I'm also okay with just exposing the boolean flag (just [this](https://github.com/microsoft/fluentui-react-native/pull/1931/commits/3f1fcb97191d42b1df0fe84ced503bbb723a6875) commit).

**Comment by rurikoaraki** [comment]
@nakambo okay let me think about it a little bit, i'll get back to the PR later today. If we do decide to add the full functionality we should update the documentation to say that this is the behavior of specifying custom key handlers.

**Review by Saadnajmi — APPROVED** [review]
Good from my side, but I would wait for final approval from @rurikoaraki.

For what it's worth, my thoughts overall are:

1) If we can get away with just exporting the one flag and not changing behavior, I prefer that. 
2) If we do change behavior, I would document it (probably Buttons' SPEC.md file). 
3) I'd normally push for a more flexible API, but given the priority of this bug / the level of refactoring needed to do that / timeline, I don't think that's feasible at this point.

**Comment by rurikoaraki** [comment]
@nakambo Okay given that preventDefault doesn't work properly this change makes sense to me. Could add a note in documentation about this behavior?

Follow up on our side would be to make this behavior consistent across our components.

**Comment by nakambo** [comment]
> @nakambo Okay given that preventDefault doesn't work properly this change makes sense to me. Could you add a note in documentation about this behavior?
> 
> Follow up on our side would be to make this behavior consistent across our components.

I have added a note -- if that isn't what you were expecting to see, let me know specifically which file\what note you'd like to see.

**Comment by rurikoaraki** [comment]
> > @nakambo Okay given that preventDefault doesn't work properly this change makes sense to me. Could you add a note in documentation about this behavior?
> > Follow up on our side would be to make this behavior consistent across our components.
> 
> I have added a note -- if that isn't what you were expecting to see, let me know specifically which file\what note you'd like to see.

Awesome, thanks!

**Review by rurikoaraki — APPROVED** [review]

**Comment by Ruriko** [comment]
@Saadnajmi next time ping the correct person

**Pull request merged**
